lua/nvim-tree/actions/node/open-file.lua

```diff
@@ -234,7 +234,12 @@ local function open_in_new_window(filename, mode, win_ids)
     end
   end
 
-  local fname = vim.fn.fnameescape(filename)
+  local fname = vim.fn.fnameescape(utils.canonical_path(filename))
+  fname = vim.loop.fs_realpath(fname);
+
+  if utils.is_windows then
+    fname =  fname:gsub("%(", "\\(")
```

Comment:
Will we need this any more? The idea is for `fs_realpath` to escape these.

Also, we should move `fs_realpath` into `canonical_path`

Reply:
- `vim.loop.fs_realpath` did not solve the issue. If you print the path, it looks fine, but Vim still opens the wrong thing. 
It is not an issue with the module per se.  
The issue happens after/within the `pcall` function call  on line 262 
An escape of  `\(` seems nessary. 

- I checked for `[` too, but those work fine without escape.

Reply:
That is reasonable. I was naive in hoping that fs_realpath would do the job...

Let's put that gsub inside `canonical_path` behind the windows feature flag. If we discover any other such problem characters we can add them in that location.

Reply:
> The issue happens after/within the pcall function call on line 262
An escape of \( seems nessary.

If you are willing to do some more tests, would you mind replacing that string manipulation there and use

Reply:
`nvim_cmd` requires some more rewrites but it delivers the same result as using `pcall`
In the End it seems to be just another way to call native VIM functions.

Using `nvim_parse_cmd` seems to make things worse by opening a 'unknown' file.

```lua

  --if utils.is_windows then
  --  -- fname =  fname:gsub("%(", "\\(")
  --end

  local cmd
  if create_new_window then
    cmd = new_window_side .. ' vsplit'
    --cmd = string.format("%s vsplit %s", new_window_side, fname)
  elseif mode:match "split$" then
    cmd = mode
    --cmd = string.format("%s %s", mode, fname)
  else
    cmd = 'edit'
    --cmd = string.format("edit %s", fname)
  end

  if mode == "preview" and view.View.float.enable then
    -- ignore "WinLeave" autocmd on preview
    -- because the registered "WinLeave"
    -- will kill the floating window immediately
    set_current_win_no_autocmd(target_winid, { "WinLeave", "BufEnter" })
  else
    set_current_win_no_autocmd(target_winid, { "BufEnter" })
  end

  vim.api.nvim_cmd({cmd = cmd, args = {fname}}, {})
  --pcall(vim.cmd, cmd)
  lib.set_target_win()
```

Having a   `(` in the name is nothing unusual/illegal. That is why none of the escape functions react to it.
I don't understand why Vim is handling `\(` that way. What kind of function do the escaped parentheses serve?
Google does not say anything about it. 

could it be a bug in https://github.com/neovim/neovim ?